xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/ResetPassword.xml

```diff
@@ -73,6 +73,15 @@ u = user account sent in the form
   ## Check if the user exists and has a valid email address configured in his profile
   #set ($userObj = '')
   #set ($userObj = $userDoc.getObject('XWiki.XWikiUsers'))
+  ## If local user does not exist check global user
+  #if (!$userObj)
```

Comment:
Same comment about checking if we are not already on the main wiki, to avoid unnecessary work.